QuizApp/QuizApp/UI/User/UserViewController.swift

```diff
@@ -51,10 +70,37 @@ class UserViewController: UIViewController {
 
     private func addActions() {
         logoutButton.addTarget(self, action: #selector(tappedLogoutButton), for: .touchUpInside)
+        saveButton.addTarget(self, action: #selector(tappedSaveButton), for: .touchUpInside)
     }
 
     @objc private func tappedLogoutButton() {
-        viewModel.onButtonClick()
+        viewModel.onLogoutButtonClick()
+    }
+
+    @objc private func tappedSaveButton() {
+        nameTextField.resignFirstResponder()
+
+        guard let text = nameTextField.text else { return }
+
+        viewModel.onSaveButtonClick(name: text)
+    }
+
+    private func getData() {
+        viewModel.getData()
+    }
+
+    private func bindViewModel() {
+        viewModel
+            .$account
+            .sink { [weak self] account in
+                guard let self = self else { return }
+
+                DispatchQueue.main.async {
```

Comment:
All field changes from VM should be already on main. So, move this to VM or use @MainActor above VM function where you assign a new value to the account field.